feat(filesPage): rework files page ui AP-603

josephmcg opened the pull request:
<!--  Thanks for sending a pull request!
If this is your first time, please read our contributor guidelines: https://github.com/Satellite-im/Core-PWA/wiki/Contributing
-->

**What this PR does** 📖
- rework files page UI
- use flex instead of floats for layout
- clean up css

**Which issue(s) this PR fixes** 🔨
AP-603

<!--AP-X-->

**Special notes for reviewers** 🗒️
- still needs translations, I was waiting for locales to be brought into main repo first
- I'm not using the same icons as the mockup, should I rip them from Figma and add to our icon library?
- I noticed some global style inconsistencies between mockup and our app.
  - buttons and inputs are 34px height in mockup. in our app, buttons are 30px height, while inputs have 32px. I added temp styles to avoid problems while we reconcile this

**Additional comments** 🎤

stavares843 commented on components/views/files/controls/Controls.vue:
```diff
@@ -2,46 +2,43 @@
     FolderPlusIcon,
+    FilePlusIcon,
   },
+  // todo - best practice would be emitting rather than passing function as a prop
```
added ticket - AP-639

stavares843 commented on components/views/files/controls/Controls.vue:
```diff
@@ -2,46 +2,43 @@
+      }
+      this.input.show = !this.input.show
     },
+    // todo
```
added ticket - AP-640

stavares843 commented on components/views/files/file/File.html:
```diff
@@ -1,22 +1,32 @@
+  />
+  <div v-else class="icon-container">
+    <folder-icon v-if="item.children" size="4x" class="file-type-icon" />
+    <!-- todo - condition for zip image after confirming how type data will come through -->
```
added ticket - AP-641

molimauro commented on components/views/files/controls/Controls.less:
```diff
@@ -1,39 +1,43 @@
 }
 
-@media only screen and (max-width: 576px) {
+@media only screen and (max-width: 480px) {
```
can we create a breakpoint in `breakpoints.less` for the `480px` since it's used in several files?

molimauro commented on components/views/files/list/List.vue:
```diff
@@ -50,12 +50,12 @@ export default Vue.extend({
      */
     handle(item: FileType | Folder): void {
-      const hasChildren = ((<Folder>item).children)
+      const hasChildren = (<Folder>item).children
       if (hasChildren) {
         this.push(item)
```
I know it's not related to your code but can we add a guard here instead of if-else?

molimauro commented:
two small changes, other than that great! 💥

josephmcg commented:
@molimauro Thanks Mauro, fixed those and added some additional functionality for the sidebar. Now it will stick if the user is scrolling on a smaller desktop

![image](https://user-images.githubusercontent.com/33670767/151922420-00540ade-892b-477f-bba3-59e424e948af.png)

molimauro commented:
> @molimauro Thanks Mauro, fixed those and added some additional functionality for the sidebar. Now it will stick if the user is scrolling on a smaller desktop
> 
> Also cleaned up some rendering, using v-if rather than conditionally applying a css class w/ display:none
> 
> ![image](https://user-images.githubusercontent.com/33670767/151922420-00540ade-892b-477f-bba3-59e424e948af.png)

Ok thanks! just one thing, i tried to scroll on small desktop and if i scroll when hovering the file-right i can scroll, if i hover the files-left i cannot. Maybe am i missing something? From your screenshot i can see you're on the mobile responsive not the desktop one, is that intentional? :) thanks!

https://user-images.githubusercontent.com/50131258/151936458-0ff9fbf5-2791-43f2-a3ae-558fd48f66b8.mov

josephmcg commented:
Oh interesting, that was unintentional 😅 . I wasn't aware of that 'desktop' setting in dev tools, very cool! I'll take another look at this

josephmcg commented:
closed, reopened here
https://github.com/Satellite-im/Core-PWA/pull/1262
